feat: add link code extension (#269)
ansys/ansys-sphinx-theme: doc/source/user_guide/extensions.rst

.. _ref_user_guide_extension:

Extensions
==========

> This page contains the extensions shipped extensions that seamlessly integrate with this theme to enhance 

Reviewer: ```suggestion
This page contains the shipped extensions that seamlessly integrate with this theme to enhance 
```

> documentation functionality 

Reviewer: ```suggestion
documentation functionality.
```

Linkcode extension
-------------------

The Linkcode extension automatically adds "View Source" links to the documentation for Python, C, C++, 
and JavaScript objects. It allows you to link to the source code hosted on GitHub.

Installation
~~~~~~~~~~~~

To use the ``ansys-sphinx-theme`` linkcode extension, you need to add it to the Sphinx configuration:

#. Add the following line in the ``conf.py`` file:
   .. code-block:: python

      extensions = ["ansys_sphinx_theme.extension.linkcode"]

Configuration options
---------------------

> 

Reviewer: You should also explain the "priorities" for both config options.

If both configs are given, which one takes precedence?

Direct and explicit config should take precedence: i.e. ``link_code_library`` et al.
Clarify this in the docs please.

Author: Thats true, i added. Thanks @RobPasMue

> The Linkcode extension provides a way to configure its behavior by using certain options within your conf.py file. 

Reviewer: ```suggestion
The Linkcode extension provides a way to configure its behavior by using certain options within your ``conf.py`` file. 
```
Be consistent on formatting. Review the docs for misalignments.

Depending on your preferred approach, you can utilize the direct 

> configuration options or the html_context dictionary to streamline your settings.

Reviewer: ```suggestion
configuration options or the ``html_context`` dictionary to streamline your settings.
```

> Direct configuration options:
> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer: ```suggestion
Direct configuration options
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

- ``link_code_library`` (str, optional):
  The user/repository name where the source code is hosted. For example, ``ansys/ansys-sphinx-theme``.

- ``link_code_source`` (str, optional):
  The relative path of the source code file within the repository. For example, ``src``.

- ``link_code_branch`` (str, optional, default: 'main'):
  The GitHub branch. It can be a specific version like ``main`` or ``dev``.

> Using ``html_context`` dictionary:
> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer: ```suggestion
Using ``html_context`` dictionary
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

You also have the option to centralize your GitHub-related configuration by incorporating it 
directly into the ``html_context`` dictionary within your `conf.py` file. This approach allows you to 
minimize redundancy and manage the GitHub-related information more effectively for the extension:

.. code-block:: python

   # Example of setting GitHub-related configuration in conf.py
   html_context = {
       # "github_url": "https://github.com", # or your GitHub Enterprise site
       "github_user": "<your-github-org>",
       "github_repo": "<your-github-repo>",
       "github_version": "<your-branch>",
       "source_path": "<path-from-root-to-your-source_file>",
   }

With this setup, you can fine-tune your configuration according to your preferences and requirements, 

> enhancing the integration of the Linkcode extension into your documentation.

Reviewer: ```suggestion
enhancing the integration of the ``linkcode`` extension into your documentation.
```

I am seeing Linkcode, linkcode, ... Please stick with one of them throughout the docs. I suggest using ``linkcode`` (i.e. formatted as code)

.. _ref_user_guide_extension:

Extensions
==========

This page contains the shipped extensions that seamlessly integrate with this theme to enhance 
documentation functionality.

Linkcode extension
-------------------

The Linkcode extension automatically adds "Source" links to the documentation for Python, C, C++, 
and JavaScript objects. It allows you to link to the source code hosted on GitHub.

Installation
~~~~~~~~~~~~

To use the ``ansys-sphinx-theme`` linkcode extension, you need to add it to the Sphinx configuration:

#. Add the following line in the ``conf.py`` file:
  .. code-block:: python

    extensions = ["ansys_sphinx_theme.extension.linkcode"]

Configuration options
---------------------

The Linkcode extension provides a way to configure its behavior by using certain options within your ``conf.py`` file. 
Depending on your preferred approach, you can utilize the direct 
configuration options or the ``html_context`` dictionary to streamline your settings.

> If both sets of configuration options are given, the direct configuration options (example, ``link_code_library``,

Reviewer: ```suggestion
If both sets of configuration options are given, the direct configuration options (that is, ``link_code_library``,
```

``link_code_source``, ``link_code_branch``) has
precedence over the corresponding settings in the ``html_context`` dictionary.

Direct configuration options
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

- ``link_code_library`` :
  The user/repository name where the source code is hosted. For example, ``ansys/ansys-sphinx-theme``.

- ``link_code_source`` (str, optional, default: ''):
  The relative path of the source code file within the repository. For example, ``src``.

- ``link_code_branch`` (str, optional, default: 'main'):
  The GitHub branch. It can be a specific version like ``main`` or ``dev``.

If the ``link_code_source`` and ``link_code_branch`` options are not provided in the configuration, 

> the following default values will be used:

Reviewer: ```suggestion
the following default values are used:
```

> - ``link_code_source``: An empty string (``''``). This will link to the root of the repository.

Reviewer: ```suggestion
- ``link_code_source``: An empty string (``''``). This links to the root of the repository.
```
